typescript/sdk/src/ism/EvmIsmModule.hardhat-test.ts

```diff
@@ -0,0 +1,720 @@
+    multiProvider = MultiProvider.createTestMultiProvider({ signer });
+
+    ismFactoryDeployer = new HyperlaneProxyFactoryDeployer(multiProvider);
+    const contractsMap = await ismFactoryDeployer.deploy(
+      multiProvider.mapKnownChains(() => ({})),
+    );
+
+    // get addresses of factories for the chain
+    factoryContracts = contractsMap[chain];
+    factoryAddresses = Object.keys(factoryContracts).reduce((acc, key) => {
+      acc[key] =
+        contractsMap[chain][key as keyof ProxyFactoryFactories].address;
+      return acc;
+    }, {} as Record<string, Address>) as HyperlaneAddresses<ProxyFactoryFactories>;
+
+    // legacy HyperlaneIsmFactory is required to do a core deploy
+    const legacyIsmFactory = new HyperlaneIsmFactory(
+      contractsMap,
+      multiProvider,
+    );
+
+    // mailbox
+    mailboxAddress = (
+      await new TestCoreDeployer(multiProvider, legacyIsmFactory).deployApp()
+    ).getContracts(chain).mailbox.address;
+
+    // new mailbox
+    newMailboxAddress = (
+      await new TestCoreDeployer(multiProvider, legacyIsmFactory).deployApp()
+    ).getContracts(chain).mailbox.address;
+
+    // example routing config
+    exampleRoutingConfig = {
+      type: IsmType.ROUTING,
+      owner: await multiProvider.getSignerAddress(chain),
+      domains: Object.fromEntries(
+        testChains
+          .filter((c) => c !== TestChainName.test4)
+          .map((c) => [c, randomMultisigIsmConfig(3, 5)]),
+      ),
+    };
+  });
+
+  // Wrapper for creating an ISM module
+  // include option for mailbox address override
+  async function createIsmModule({
+    config,
+    mailbox = mailboxAddress,
+  }: {
+    config: IsmConfig;
+    mailbox?: Address;
+  }): Promise<EvmIsmModule> {
+    return EvmIsmModule.create({
+      chain,
+      config,
+      deployer: ismFactoryDeployer,
+      factories: factoryAddresses,
+      mailbox,
+      multiProvider,
+    });
+  }
+
+  // Wrapper for checking whether ISM module matches a given config
+  // include option for mailbox address override
+  async function ismModuleMatchesConfig({
+    ism,
+    config,
+    mailbox = mailboxAddress,
+    configured = true,
+  }: {
+    ism: EvmIsmModule;
+    config: IsmConfig;
+    mailbox?: Address;
+    configured: boolean;
+  }): Promise<boolean> {
+    return moduleMatchesConfig(
+      chain,
+      ism.serialize().deployedIsm,
+      config,
+      multiProvider,
+      factoryContracts,
+      mailbox,
+      configured,
+    );
+  }
+
+  // Wrapper for create a new multiprovider with an impersonated account
+  async function impersonateAccount(account: Address): Promise<MultiProvider> {
+    await hre.ethers.provider.send('hardhat_impersonateAccount', [account]);
+    await fundingAccount.sendTransaction({
+      to: account,
+      value: hre.ethers.utils.parseEther('1.0'),
+    });
+    return MultiProvider.createTestMultiProvider({
+      signer: hre.ethers.provider.getSigner(account),
+    });
+  }
```

Comment:
from our sync: https://github.com/hyperlane-xyz/hyperlane-monorepo/blob/bd6ca01fd4dd4c6d8491c36b633308829008c8f9/typescript/sdk/src/utils/fork.ts#L70

Reply:
existing one is for anvil and doesn't quite work in our hh test, leaving for now and will extract out when required in future test

Reply:
anvil aliases the hardhat RPCs so should be compatible fwiw

Reply:
I did try using the existing ones and even writing a couple of `hardhat_XYZ` helpers, but I think the issue is that because it's hh unit testing, you have to go via `hre.ethers.provider`